rxkeystore/src/main/java/com/nexenio/rxkeystore/provider/cipher/symmetric/RxSymmetricCipherProvider.java

```diff
@@ -17,6 +20,11 @@
 
     Single<SecretKey> generateKey(@NonNull String alias, @NonNull Context context);
 
+    /**
+     * Note: Depending on the type and provider used in {@link KeyStore(KeyStoreSpi, Provider, String)}, the purposes may be ignored.
+     */
+    Single<SecretKey> generateKey(@NonNull String alias, @Nullable Integer keyPurposes, @NonNull Context context);
```

Comment:
I wouldn't make keyPurposes nullable. Instead, you could provide the properties where you pass null to this method.